.circleci/deploy.sh

```diff
@@ -47,14 +49,45 @@ sudo pip install --upgrade pip
 # Install awscli to make "aws" command available
 sudo pip install awscli
 
-# Move files in S3 buckets from temporary to permanent locations.
-aws s3 sync submitted_data/  s3://cimr-root/PR-${PR_NUMBER}/ --exclude "request.handled"
-aws s3 sync processed_data/  s3://cimr-d/
+# Copy "processed_data" to "cimr-d" bucket (public).
+# "PR-<n>" is inserted in filenames to avoid duplicates.
+OUTPUT_FILES=$(find processed_data -type f)
+for f in ${OUTPUT_FILES}; do
+    g=$(echo $f | cut -d'/' -f'2-')    # strip "processed_data/" from $f
+    S3DIR=$(dirname $g)                # S3 object's dir name
+
+    g_base=$(basename $g)              # base filename
+    tokens=(${g_base//./ })            # split $g_base into an array using '/'
+    LEN="${#tokens[@]}"                # number of items in tokens
+
+    if [ $LEN -eq 1 ]; then
+	# "foo" will become "foo-PR-n"
+	S3BASE="${g_base}-${PR_STR}"
+    elif [ $LEN -eq 2 ]; then
+	# "foo.ext1" will become "foo-PR-n.ext1"
+	S3BASE="${tokens[0]}-${PR_STR}.${tokens[1]}"
+    else
+	# "foo.ext2.ext1" will become "foo-PR-n.ext2.ext1"
```

Comment:
This only works if len > 2 indicates two extensions, which is fine in this case. I just need to make sure this will be true for other files cimr will accept in future versions.

Reply:
As you can see, if the output file is "foo.txt", it will become "foo-PR-n.txt" in S3 bucket.